### ISSUE TITLE
Add support for system properties not defined in pom.xml for the gradle cache

### ISSUE THREAD

**Issue by vmassol** [gradle/gradle#20109]
For example the surefire plugin handler for the gradle cache only support system properties defined as https://maven.apache.org/surefire/maven-surefire-plugin/examples/system-properties.html

However, in Maven, you only need to add that to the pom if you need to pass systel properties from the POM to the tests. If you pass them on the maven command line, the tests can access them and there's no need to change the pom.xml files.

It would be great if the gradle cache for Maven could support this too, maybe with some options to configure them as inputs (https://docs.gradle.com/enterprise/maven-extension/#declaring_inputs_and_outputs) if they cannot be found automatically.

Thanks!

Context: the XWiki build

**Comment by vmassol** [comment]
Note: it also forces to always have the system properties defined whereas java code can expect the properties to not exist and then fallback on default values for example.

**Comment by vlsi** [comment]
@vmassol , would you please clarify what goal are you trying to achieve?
What is the exact use case?

**Comment by vmassol** [comment]
Thanks @vlsi. So we're trying to be able to use the gradle cache for the XWiki open source project and especially to be able to use the cache for functional tests. These tests are executed with different maven command lines, specifying the environment to run them in (see https://dev.xwiki.org/xwiki/bin/view/Community/Testing/DockerTesting/). We've noticed that Gradle currently wrongly caches the test results even though we pass different system properties (`-Dxxx=yyy`). That's because currently only system properties defined in the surefire plugin configuration are supported as cache inputs.

Note that I've tried this morning to add our command line system properties inside the pom.xml files but it's not possible since we support dynamic properties (e.g. `-Dxwiki.test.ui.properties.*=xxx`)  and it's not possible to list all possible combinations (too much work).

Note: I'm in contact with @alextu
